src/openms/include/OpenMS/QC/MQEvidenceExporter.h

```diff
@@ -0,0 +1,178 @@
+
+      @param cmap ConsensusMap that includes ConsensusFeatures
+
+      @return Returns map, the index is a FeatureID, the value is the index of the ConsensusFeature
+      in the vector of ConsensusMap
+    */
+    std::map<OpenMS::Size, OpenMS::Size> makeFeatureUIDtoConsensusMapIndex_(const OpenMS::ConsensusMap &cmap);
+
+    /**
+      @brief Checks if Feature has valid PeptideIdentifications
+
+        If there are no PeptideIdentifications or the best hit of the Feature cannot be found in corresponding ConsensusFeature,
+        the functions returns false to show that something went wrong.
+
+      @param cmap, c_feature_number, UIDs and mp_f for comparing Feature and ConsensusFeature, f is used to extract PeptideIdentifications
+
+      @return Returns true if the PeptideIdentifications are valid
+    */
+    bool hasValidPepID_(
+            const OpenMS::Feature &f,
+            const OpenMS::Size c_feature_number,
+            const std::multimap<OpenMS::String, std::pair<OpenMS::Size, OpenMS::Size>> &UIDs,
+            const OpenMS::ProteinIdentification::Mapping &mp_f);
+
+    /**
+      @brief Checks if ConsensusFeature has valid PeptideIdentifications
+
+      If there are no PeptideIdentifications,
+      the functions returns false to show that something went wrong.
+
+      @param cf is used to extract PeptideIdentifications
+
+      @return Returns true if the PeptideIdentifications are valid
+    */
+    bool hasPeptideIdentifications_(const OpenMS::ConsensusFeature &cf);
+
+    /**
+      @brief Export one Feature as a row in MQEvidence.txt
+
+        Export one Feature as a row in MQEvidence.txt.
+        hasValidPepID and HasPeptideIdentifications are used.
+        If there are problems (missing data or confusing data), no output will be generated.
+
+      @param f, cmap and c_number_feature to extract data out of Feature or ConsensusFeature,
+      raw_file is specifying the raw_file the feature belongs to,
+      c_feature_number, mp_f and UIDs are used in hasValidPepID
+
+    */
+    void exportRowFromFeature_(
+            const OpenMS::Feature &f,
+            const OpenMS::ConsensusMap &cmap,
+            const OpenMS::Size c_feature_number,
+            const OpenMS::String &raw_file,
+            const std::multimap<OpenMS::String, std::pair<OpenMS::Size, OpenMS::Size>> &UIDs,
+            const OpenMS::ProteinIdentification::Mapping &mp_f);
+
+public:
+/**
+  @brief Creates MQEvidence object and file MQEvidence.txt in given path
+
+        If the path for the constructor is empty (path not valid), no file is created.
+        If the creation of the fstream object is successful a constant header is added to the file
+
+  @param path that is the path where file has to be stored
+
+*/
+    explicit MQEvidence(const OpenMS::String &path);
+
+    /**
+      @brief Closes f_stream and destruct MQEvidence object
+    */
+    ~MQEvidence();
+
+    /**
+        @brief Checks if file is writable
```

Comment:
checks if evidence.txt is writable (i.e. the path in the ctor was not empty and could be created)